src/test/java/no/ndla/taxonomy/rest/v1/NodesTest.java

```diff
@@ -1113,6 +1115,151 @@ public void update_and_deletion_of_quality_evaluation_does_not_break_calculation
+        assertEquals(3, topic2.getChildQualityEvaluationAverage().get().getCount());
+        assertEquals(
+                3.3333333333333335,
+                topic2.getChildQualityEvaluationAverage().get().getAverageValue());
+
+        var subject1 = nodeRepository.getByPublicId(URI.create("urn:subject:1"));
+        assertTrue(subject1.getChildQualityEvaluationAverage().isPresent());
+        assertEquals(6, subject1.getChildQualityEvaluationAverage().get().getCount());
+        assertEquals(
+                3.3333333333333335,
+                subject1.getChildQualityEvaluationAverage().get().getAverageValue());
+    }
+
+    @Test
+    public void moving_a_topic_also_moves_the_quality_evaluation_calculation() throws Exception {
+        // Create structure with resources and quality evaluation
+        var s1Node = builder.node(
+                NodeType.SUBJECT,
+                s1 -> s1.name("S1").publicId("urn:subject:1").child(NodeType.TOPIC, t1 -> t1.name("T1")
+                        .publicId("urn:topic:1")
+                        .child(
+                                NodeType.RESOURCE,
+                                r1 -> r1.name("R1").publicId("urn:resource:1").qualityEvaluation(Grade.Five))
+                        .child(
+                                NodeType.RESOURCE,
+                                r2 -> r2.name("R2").publicId("urn:resource:2").qualityEvaluation(Grade.Four))
+                        .child(
+                                NodeType.RESOURCE,
+                                r3 -> r3.name("R3").publicId("urn:resource:3").qualityEvaluation(Grade.One))));
+
+        builder.node(NodeType.SUBJECT, s2 -> s2.name("S2").publicId("urn:subject:2"));
+
+        s1Node.updateEntireAverageTree();
+        var topic1 = nodeRepository.getByPublicId(URI.create("urn:topic:1"));
+        assertEquals(3, topic1.getChildNodes().size());
+        topic1.updateEntireAverageTree();
+
+        {
+            assertTrue(topic1.getChildQualityEvaluationAverage().isPresent());
+            assertEquals(3, topic1.getChildQualityEvaluationAverage().get().getCount());
+            assertEquals(
+                    3.3333333333333335,
+                    topic1.getChildQualityEvaluationAverage().get().getAverageValue());
+
+            var subject1 = nodeRepository.getByPublicId(URI.create("urn:subject:1"));
+            assertTrue(subject1.getChildQualityEvaluationAverage().isPresent());
+            assertEquals(3, subject1.getChildQualityEvaluationAverage().get().getCount());
+            assertEquals(
+                    3.3333333333333335,
+                    subject1.getChildQualityEvaluationAverage().get().getAverageValue());
+
+            var subject2 = nodeRepository.getByPublicId(URI.create("urn:subject:2"));
+            assertFalse(subject2.getChildQualityEvaluationAverage().isPresent());
+        }
+
+        { // Move urn:topic:1 to urn:subject:2
```

Comment:
Er ikkje disse blokkene avhengig av kvarandre? Då er det litt misvisende at koden er strukturert slik. Dette er vel hovedsaklig for å kunne scope variabelnavn slik at dei kan gjenbrukes.  Men har ingen praktisk betydning.

Reply:
Personlig så syns jeg det økte lesbarheten til testen, men kan ta de vekk 👍